### PR TITLE
Move create, get, and replace user to pipeline architecture

### PR DESCRIPTION
**Pull request by JoshGendein**

### CODE REVIEW

File: sdk/cosmos/src/operations/create_user.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct CreateUserBody<'a> {
+    id: &'a str,
+}
+
+#[derive(Debug, Clone)]
+pub struct CreateUserResponse {
```
Comment (JoshGendein) [inline sdk/cosmos/src/operations/create_user.rs:43]: Previously Create, Get, and Replace all shared CreateUserResponse. I separated this out into `CreateUserResponse`, `GetUserResponse`, and `ReplaceUserResponse`. Is this ok, or should there be a shared `UserResponse`?

File: sdk/cosmos/src/consistency_level.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 use crate::headers;
+use crate::prelude::CreateUserResponse;
```
Comment (JoshGendein) [inline sdk/cosmos/src/consistency_level.rs:2]: I was unsure what to do here, I noticed in Ryan's example PR for converting to pipeline he removed the implement_from! for GetDatabaseResponse but in this case it was still required by examples/permission_00 @ line 66.

**Review by rylev — APPROVED** [review]
🎉Great stuff! Thank you!

@MindFlavor could you test this and #331 and merge if it looks good to you?

File: sdk/cosmos/src/operations/create_user.rs (hunk 1 above)
Comment (JoshGendein) [reply sdk/cosmos/src/operations/create_user.rs:43]: I like the idea of sharing the response, I suppose the only downside is if the response changes in the future for one but not the others. Follow-up question, where should I put the shared `UserResponse`? There is already `cosmos/src/resources/user.rs` or I can create a new `user` module under 'resources'.

Comment (heaths) [reply sdk/cosmos/src/operations/create_user.rs:43]: For Azure SDKs we generally share requests i.e. input but not responses i.e. output. At least in OO or duck-type languages you can always pass something more derived in but getting something more derived out can be problematic (how does the caller know they can upcast safely)?

When in doubt, I'd take a look at our Azure/azure-sdk-for-net (or -java or -cpp) repos for consistency with those. Much discussion was already done about those trying to make it work for all languages as much as possible.

Comment (JoshGendein) [reply sdk/cosmos/src/operations/create_user.rs:43]: It looks like in Azure/azure-sdk-for-java they use a shared [`CosmosUserResponse`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosUserResponse.java). I couldn't find anything in Azure/azure-sdk-for-net but I did find that in Azure/azure-cosmos-dotnet-v3 they use a shared [`UserResponse`](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Resource/User/UserResponse.cs).

Comment (heaths) [reply sdk/cosmos/src/operations/create_user.rs:43]: Any repos besides Azure/azure-sdk-for-* are old and should not be used as examples. The Azure SDK team was created to unify the different languages (idiomatically) for a consistent API, among other things like mitigating breaking changes from purely REST-generated source.

In .NET, we typically return a `Response<T>`, so if you find the method you might find something like `Response<CosmosUser>` (shipping a class called simply "User" would never pass our archboard, per guidelines). Other languages do something similar.

Comment (heaths) [reply sdk/cosmos/src/operations/create_user.rs:43]: @tg-msft do you know what is ideal here? I haven't really been following all the Cosmos discussions and not even sure where to find source. Is it still track 1?

Comment (tg-msft) [reply sdk/cosmos/src/operations/create_user.rs:43]: https://apiview.dev/Assemblies/Review/0de57f00b1ad4fdab2fa218f59863c78 is the shape of the API that Krzysztof has been working with them on, but it's not GA'ed yet.  I think the source lives at https://github.com/Azure/azure-cosmos-dotnet-v3/tree/v4.

**Review by MindFlavor — APPROVED** [review]
Nice job, thank you! 
I also ran the E2E tests against my subscription and they worked flawlessly 👍 .

**Pull request merged**